stock_demand_estimate_matrix/wizards/stock_demand_estimate_wizard.py

```diff
@@ -0,0 +1,263 @@
+                                _range,
+                                product,
+                                uom_id,
+                                uom_qty,
+                                estimate_id=estimate_id,
+                            ),
+                        )
+                    )
+            sheet.line_ids = lines
+
+    def _get_ranges(self):
+        domain_1 = [
+            "&",
+            ("type_id", "=", self.date_range_type_id.id),
+            "|",
+            "&",
+            ("date_start", ">=", self.date_start),
+            ("date_start", "<=", self.date_end),
+            "&",
+            ("date_end", ">=", self.date_start),
+            ("date_end", "<=", self.date_end),
+        ]
+        domain_2 = [
+            "&",
+            ("type_id", "=", self.date_range_type_id.id),
+            "&",
+            ("date_start", "<=", self.date_start),
+            ("date_end", ">=", self.date_start),
+        ]
+        domain = expression.OR([domain_1, domain_2])
+        ranges = self.env["date.range"].search(domain)
+        return ranges
+
+    def _get_default_estimate_line(
+        self, _range, product, uom_id, uom_qty, estimate_id=None
+    ):
+        name_y = "{} - {}".format(product.name, product.uom_id.name)
+        if product.default_code:
+            name_y += "[{}] {}".format(product.default_code, name_y)
+        values = {
+            "value_x": _range.name,
+            "value_y": name_y,
+            "date_range_id": _range.id,
+            "product_id": product.id,
+            "product_uom": uom_id,
+            "product_uom_qty": uom_qty,
+            "location_id": self.location_id.id,
+            "estimate_id": estimate_id,
+        }
+        return values
+
+    @api.model
+    def _prepare_estimate_data(self, line):
+        return {
+            "date_range_id": line.date_range_id.id,
+            "product_id": line.product_id.id,
+            "location_id": line.location_id.id,
+            "product_uom_qty": line.product_uom_qty,
+            "product_uom": line.product_id.uom_id.id,
+        }
+
+    def button_validate(self):
+        res = []
+        for line in self.line_ids:
+            if line.estimate_id:
+                line.estimate_id.product_uom_qty = line.product_uom_qty
+                res.append(line.estimate_id.id)
+            else:
+                data = self._prepare_estimate_data(line)
+                estimate = self.env["stock.demand.estimate"].create(data)
+                res.append(estimate.id)
+        res = {
+            "domain": [("id", "in", res)],
+            "name": _("Stock Demand Estimates"),
+            "src_model": "stock.demand.estimate.wizard",
+            "view_type": "form",
+            "view_mode": "tree",
+            "res_model": "stock.demand.estimate",
+            "type": "ir.actions.act_window",
+        }
+        return res
+
+
+class StockDemandEstimateSheetLine(models.TransientModel):
```

Comment:
Please put one model per file

Reply:
so today it is a blocking request? :laughing:

Reply:
I don't want to sound rude, it is just that this is not a big deal and will add a bid diff in the migration as well, but we will do it time and we will correct the commit history as well.

Thanks for the time you spend reviewing.

Reply:
Done!